### ISSUE TITLE
Charging state: charging / not charging 

### ISSUE THREAD

**Issue by traxata** [Hyundai-Kia-Connect/kia_uvo#947]
I appreciate that the 2024 vehicles are in a bit of a limbo state feature wise at the moment but I am wondering if there's an entity in the API that shows a binary state for if your vehicle is charging or not?
 I have one for plugged in and if anything is connected to the charge port but I would like to know if the vehicle is charging as I have a bunch of home batteries that are set up with automations based on if certain things are charging or not. I have this set up with my old vehicle (polestar 2), portable ecoflow batteries (charging via CCS2 connectors) and a Renault Zoë. My logic is checking to see if the Ohme wall charger is charging and then what device it is charging before doing anything. 

This allows me to do safety checks (to make sure no one is charging who shouldn't be) and also trigger pauses and unpauses to charging for Intelligent Octopus Energy smart tariffs based on cheap electricity slots. 

So basically all I'm looking for is a charging Yes / no sensor state anything else is a bonus but not needed for my use cases :)

**Comment by erelke** [comment]
binary_sensor.******_ev_battery_charge

**Comment by cdnninja** [comment]
What data shows for you today?

**Comment by cdnninja** [comment]
Also please check your data sensor for extended attributes to see if something in that answers this. If so can easily expose it.

**Comment by traxata** [comment]
![Screenshot_20241017-064452](https://github.com/user-attachments/assets/1be93d82-a417-48a8-baa0-f23e4a821d74)
I do not have an EV battery charge with the EV9.

Will try and check if there's any extended that I'm missing by default

**Comment by traxata** [comment]
I see that 

`HyundaiKiaBinarySensorEntityDescription(
        key="ev_battery_is_charging",
        name="EV Battery Charge",
        is_on=lambda vehicle: vehicle.ev_battery_is_charging,
        device_class=BinarySensorDeviceClass.BATTERY_CHARGING,
    ),`

Exists in the binary sensors.py but I don't have that sensor showing through to me in the entity in HA

**Comment by cdnninja** [comment]
That implies vehicle.ev_battery_is_charging isn't populated.  Need to dig into your raw data from the api and look at the mapping to see why.   Extended attributes show the raw data.

**Comment by traxata** [comment]
Ok thank you. 
![Screenshot_20241031-064940](https://github.com/user-attachments/assets/790fbcb2-c1b0-4d3f-ab1e-e235111e535a)
Weirdly the car card shows this but I think it must be calculating that from other sensors.

I cannot seem to work out how to show the extended attributes, would you be able to point me in the right direction? Then happy to show you what it says 😅

**Comment by cdnninja** [comment]
Go Developer Menu>States>select the "data" entity from this integration.  It should show a list below of all sorts of data.

**Comment by cdnninja** [comment]
Debug logs work for this too.

**Comment by traxata** [comment]
So I made a work around instead in case anyone else has the issue:

template:
  - binary_sensor:
      - name: "EV9 Charging Status"
        state: >
          {% set ev_port_open = is_state('binary_sensor.ev9_ev_charge_port', 'on') %}
          {% set ev_plugged = is_state('binary_sensor.ev9_ev_battery_plug', 'on') %}
          {% set ohme_connected = is_state('binary_sensor.ohme_car_connected', 'on') %}
          {% set ohme_charging = is_state('binary_sensor.ohme_car_charging', 'on') %}
          {% set power_draw = states('sensor.ohme_power_draw') | float(0) %}
          {% set battery = states('sensor.ev9_ev_battery_level') | float(0) %}
          {% set previous_battery = state_attr('sensor.ev9_ev_battery_level', 'last_value') | float(0) %}

          {{ ev_port_open and ev_plugged and ohme_connected and ohme_charging and power_draw > 500 and battery > previous_battery }}
        icon: >
          {% if ev_port_open and ev_plugged and ohme_connected and ohme_charging and power_draw > 500 and battery > previous_battery %}
            mdi:ev-station
          {% else %}
            mdi:battery
          {% endif %}

In this order:
Looks to see if charging flap is open. Looks to see if car is plugged in. Then Looking to see if the Ohme car charger is connected, then charging and that the power flowing is higher than 500W.
Then if the SOC level is rising.


It looks at all of these as I plan to have a quasar 2 bidirectional (V2H) charger and I use the V2L adapter when out and about in the car

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
